lib/lightning/apollo_client/sse_stream.ex

```diff
@@ -0,0 +1,288 @@
+defmodule Lightning.ApolloClient.SSEStream do
+  @moduledoc """
+  Server-Sent Events (SSE) client for streaming AI responses from Apollo server.
+
+  This module handles HTTP streaming connections to Apollo's SSE endpoints,
+  parsing incoming events and forwarding them to the appropriate channels.
+  """
+  use GenServer
+
+  require Logger
+
+  @doc """
+  Starts a streaming SSE connection to Apollo server.
+
+  ## Parameters
+
+  - `url` - HTTP URL for Apollo streaming endpoint
+  - `payload` - Request payload to send to Apollo
+
+  ## Returns
+
+  - `{:ok, pid}` - SSE stream process started successfully
+  - `{:error, reason}` - Failed to establish connection
+  """
+  def start_stream(url, payload) do
+    GenServer.start_link(__MODULE__, {url, payload})
+  end
+
+  @impl GenServer
+  def init({url, payload}) do
+    lightning_session_id = payload["lightning_session_id"]
+    apollo_payload = Map.delete(payload, "lightning_session_id")
+
+    apollo_timeout = Lightning.Config.apollo(:timeout) || 30_000
+    stream_timeout = apollo_timeout + 10_000
+
+    timeout_ref = Process.send_after(self(), :stream_timeout, stream_timeout)
+
+    parent = self()
+
+    spawn_link(fn ->
+      stream_request(url, apollo_payload, parent, lightning_session_id)
+    end)
+
+    {:ok,
+     %{
+       session_id: lightning_session_id,
+       timeout_ref: timeout_ref,
+       completed: false
+     }}
+  end
+
+  @impl GenServer
+  def handle_info({:sse_event, event_type, data}, state) do
+    handle_sse_event(event_type, data, state)
+    {:noreply, state}
+  end
+
+  def handle_info(:stream_timeout, %{completed: false} = state) do
+    Logger.error("[SSEStream] Stream timeout for session #{state.session_id}")
+    broadcast_error(state.session_id, "Request timed out. Please try again.")
+    {:stop, :timeout, state}
+  end
+
+  def handle_info(:stream_timeout, state) do
+    {:noreply, state}
+  end
+
+  def handle_info({:sse_complete}, state) do
+    if state.timeout_ref, do: Process.cancel_timer(state.timeout_ref)
+    Logger.info("[SSEStream] Stream completed for session #{state.session_id}")
+    {:stop, :normal, %{state | completed: true}}
+  end
+
+  def handle_info({:sse_error, reason}, state) do
+    if state.timeout_ref, do: Process.cancel_timer(state.timeout_ref)
+
+    Logger.error(
+      "[SSEStream] Stream error for session #{state.session_id}: #{inspect(reason)}"
+    )
+
+    error_message =
+      case reason do
+        :timeout -> "Connection timed out"
+        :closed -> "Connection closed unexpectedly"
+        {:shutdown, _} -> "Server shut down"
+        {:http_error, status} -> "Server returned error status #{status}"
+        _ -> "Connection error: #{inspect(reason)}"
```

Comment:
This looks like this:

<img width="565" height="123" alt="image" src="https://github.com/user-attachments/assets/b74d795e-3660-4710-96d6-57e52c2a34b0" />


Are you able to just print that reason string?